File: Data Structures and Algorithms/LeetCode/121. Best Time to Buy and Sell Stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
     int profit = 0, bought = prices[0];
       
        for (int i=1; i<prices.Length; i++)
        {
            if(prices[i] < bought) bought = prices[i];
            else if (profit < prices[i] - bought) profit = prices[i] - bought;
        }
        
        return profit;
    }
}

File: Data Structures and Algorithms/LeetCode/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mind = 100000
        profit = 0
        for i in range(len(prices)):
            if prices[i] < mind:
                mind = prices[i]
            profit = max(prices[i]-mind, profit)
        return profit

    # What we are doing here is simply keeping track of the minimum price we have seen so far and the maximum profit we can make.
